### PR TITLE
Remove needless output while `rake test`

### PR DESCRIPTION
**Pull request by tadd**
`rake test` prints needless and redundant strings currently. This PR tries to remove all of them.

Before:

```
$ bundle exec rake test 
/home/tadashi/git/rbs/test/rbs/test/runtime_test_test.rb:108: warning: assigned but unused variable - output
/home/tadashi/git/rbs/test/rbs/test/runtime_test_test.rb:134: warning: ambiguous first argument; put parentheses or a space even after `/' operator

# Running tests with run options --seed 1919:

.......................W, [2020-09-16T11:20:23.082667 #513922]  WARN -- rbs: `incompatible` method attribute is deprecated and ignored.
......W, [2020-09-16T11:20:23.084430 #513922]  WARN -- rbs: `overload def` syntax is deprecated. Use `...` syntax instead.
......................................................................................BSDL	      COPYING  Gemfile.lock  Rakefile	bin   exe	     lib	  schema  stdlib  test
CHANGELOG.md  Gemfile  README.md     Steepfile	docs  goodcheck.yml  rbs.gemspec  sig	  steep   vendor
ruby 2.7.1p118 (2020-08-07 revision e9e4f8430a) [x86_64-linux]
rbs 0.12.1
D, [2020-09-16T11:20:37.225065 #513984] DEBUG -- : No type checker was installed!
rake build            # Build rbs-0.12.1.gem into the pkg directory
rake clean            # Remove any temporary products
rake clobber          # Remove any generated files
rake install          # Build and install rbs-0.12.1.gem into system gems
rake install:local    # Build and install rbs-0.12.1.gem into system gems without network access
rake release[remote]  # Create tag v0.12.1 and build and push rbs-0.12.1.gem to rubygems.org
rake test             # Run tests
D, [2020-09-16T11:20:37.819015 #513986] DEBUG -- : No type checker was installed!
Traceback (most recent call last):
	12: from /home/tadashi/git/rbs/vendor/bundle/ruby/2.7.0/bin/rbs:23:in `<main>'
	11: from /home/tadashi/git/rbs/vendor/bundle/ruby/2.7.0/bin/rbs:23:in `load'
	10: from /home/tadashi/git/rbs/exe/rbs:5:in `<top (required)>'
	 9: from /home/tadashi/git/rbs/exe/rbs:5:in `require'
	 8: from /home/tadashi/git/rbs/lib/rbs/cli.rb:4:in `<top (required)>'
	 7: from /home/tadashi/git/rbs/lib/rbs/cli.rb:5:in `<module:RBS>'
	 6: from /home/tadashi/git/rbs/lib/rbs/test/setup.rb:64:in `block in <top (required)>'
	 5: from /home/tadashi/git/rbs/lib/rbs/test/setup.rb:64:in `any?'
	 4: from /home/tadashi/git/rbs/lib/rbs/test/setup.rb:64:in `block (2 levels) in <top (required)>'
	 3: from /home/tadashi/git/rbs/lib/rbs/test/setup.rb:53:in `to_absolute_typename'
	 2: from /home/tadashi/git/rbs/lib/rbs/factory.rb:8:in `type_name'
	 1: from /home/tadashi/git/rbs/lib/rbs/factory.rb:8:in `new'
/home/tadashi/git/rbs/lib/rbs/type_name.rb:19:in `initialize': unhandled exception
..........................................................................................................................

Finished tests in 25.143859s, 9.4258 tests/s, 196.9069 assertions/s.


237 tests, 4951 assertions, 0 failures, 0 errors, 0 skips
```

After:

```
$ bundle exec rake test 

# Running tests with run options --seed 27346:

.............................................................................................................................................................................................................................................

Finished tests in 25.253236s, 9.3849 tests/s, 196.1333 assertions/s.


237 tests, 4953 assertions, 0 failures, 0 errors, 0 skips
```

Because these are almost cosmetic changes for developers of RBS itself, there is no hurry to merge of course.

### CODE REVIEW

File: lib/rbs/cli.rb [hunk 1]
```diff
@@ -808,8 +809,10 @@ def run_test(args, options)
         'RBS_TEST_TARGET' => (targets.join(',') unless targets.empty?)
       }
 
-      system(env_hash, *args)
-      $?
+      out, err, status = Open3.capture3(env_hash, *args)
+      stdout.print(out)
+      stderr.print(err)
+      status
```
Comment (tadd) [inline lib/rbs/cli.rb:815]: This is the only point to change the behavior of the implementation. Calling `system` doesn't respect `@stdout` and `@stderr` variables, so I changed to use `Open3.caputure3`. This change also leads to suppress outputs of `rbs test` while testing.

File: test/rbs/test/runtime_test_test.rb [hunk 2]
```diff
@@ -112,6 +112,7 @@ def self.name
   end
 end
 RUBY
+    assert_match "No type checker was installed!", output
```
Comment (tadd) [inline test/rbs/test/runtime_test_test.rb:115]: This variable `output` was not used.

File: test/rbs/test/runtime_test_test.rb [hunk 3]
```diff
@@ -131,7 +132,7 @@ def world: () -> void
 end
 RBS
 
-    assert_match /TypeError: \[Hello#world\]/, output
+    assert_match(/TypeError: \[Hello#world\]/, output)
```
Comment (tadd) [inline test/rbs/test/runtime_test_test.rb:135]: This change eliminates the "ambiguous first argument" warning.

**Review by soutaro — APPROVED** [review]
👍 🗑️

**Pull request merged**